syntax = "proto3";

package mcutils;

service MeshCore {
  rpc subscribe (SubscribeRequest) returns (stream Event);
  rpc command (CommandRequest) returns (CommandReply);
}

message Event {
  string json = 1;
}

message CommandRequest {
  string command = 1;
  string json_args = 2;
  string json_kwargs = 3;
}
message CommandReply {
  Event event = 1;
}

message SubscribeRequest {}
